src/System.Data.SqlClient/src/System/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -99,6 +99,8 @@ public void AssertUnrecoverableStateCountIsCorrect()
 
     sealed internal class SqlInternalConnectionTds : SqlInternalConnection, IDisposable
```

Comment:
Why have you changed this lower level connection class for `SqlTransaction`? I recommend adding diagnostics in the SqlTransaction class.  Please take a look at https://github.com/dotnet/corefx/pull/8674 where the Diagnostics were added to SqlConnection and SqlCommand. 

Since you are trying to get diagnostic information for the Transactions, it is advisable to wrap the public API implementations in the diagnostic listener. 
Adding code at such low level classes is not good for two reasons
1. The function may be reused and unnecessary diagnostics are emitted.
2. You miss out on the diagnostics information because the calls from `SqlTransaction` leading to `SqlInternalConnectionTds` were not covered.